virttest/qemu_devices/qcontainer.py

```diff
@@ -2548,9 +2549,13 @@ def dimm_device_define_by_params(self, params, name):
         params = params.object_params("dimm")
         dimm_type = "nvdimm" if params.get("nv_backend") else "pc-dimm"
         attrs = qdevices.Dimm.__attributes__[:]
-        params = params.copy_from_keys(attrs)
-        dev = qdevices.Dimm(params=params, dimm_type=dimm_type)
+        dev = qdevices.Dimm(params=params.copy_from_keys(attrs),
+                            dimm_type=dimm_type)
         dev.set_param("id", "%s-%s" % ("dimm", name))
+        if dimm_type == "nvdimm" and params.get_boolean("nvdimm_set_uuid"):
+            dev.set_param("uuid", uuid.uuid5(uuid.NAMESPACE_OID, name))
```

Comment:
Shall we let users specify uuid to nvdimm devices? Actually I suggest making the new param to be string type therefore it would support both user specified value and random ones.

Reply:
Make senses, users should able to control the UUID by themself.
I refer to the documentation of [rfc4122](https://www.ietf.org/rfc/rfc4122.txt), now this param supports UUID as a URN format, if user specified a wrong format, will generate a random UUID.